### PR TITLE
lib/babe: implement secondary slot authorship verification

### LINKED ISSUE #1327: implement BABE secondary slot verification
## Task summary

- implement verification of `BabeSecondaryPlainPreDigest` ie. that block producer was authorized for that secondary slot

## Other information and links

- see https://github.com/paritytech/substrate/blob/master/client/consensus/babe/src/authorship.rs

### PR DESCRIPTION
**Pull request by noot**
## Changes

- implement verification of `BabeSecondaryPlainPreDigest` and `BabeSecondaryVRFPreDigest`

## Tests

```
go test ./lib/babe
```

also tested with kusama,  please see instructions here https://github.com/ChainSafe/gossamer/pull/1312

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- closes #1327

### CODE REVIEW

File: lib/babe/secondary_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		err = verifySecondarySlotPlain(uint32(i), 77, numAuths, Randomness{})
+		if err == nil {
+			numAuthorized++
+		}
+	}
+
+	require.Equal(t, 1, numAuthorized, "only one block producer should be authorized per secondary slot")
+}
+
+func createSecondaryVRFPreDigest(t *testing.T, keypair *sr25519.Keypair, index uint32, slot, epoch uint64, randomness Randomness) *types.BabeSecondaryVRFPreDigest {
```
Comment (RyRy79261) [inline lib/babe/secondary_test.go:46]: Should `slot` have a type here?

**Review by RyRy79261 — APPROVED** [review]

File: lib/babe/secondary_test.go (hunk 1 above)
Comment (noot) [reply lib/babe/secondary_test.go:46]: `uint64` is the type, for parameters with the same type you can just put the type once at the end

**Comment by noot** [comment]
@edwardmack could you please pull again and retry? seems something got messed up when i merged development, should be working now

**Review by edwardmack — APPROVED** [review]
Nice job, it's syncing now.

**Pull request merged**